containers/wordpress/run.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+# Copyright (C) 2021 Intel Corporation
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files
+# (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge,
+# publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+# OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+# OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# SPDX-License-Identifier: MIT
+
+COUNT=${COUNT:-6}
+IMAGE_NAME=${IMAGE_NAME:-"wp_base_http"}
+NUMA_PINNING=${NUMA_PINNING:-0}
+
+while test $# -gt 0; do
+    case "$1" in
+    -h | --help)
+        echo "run.sh - run workload and calculate total TPS"
+        echo " "
+        echo "run.sh [options]"
+        echo " "
+        echo "options:"
+        echo "-h, --help                 show brief help"
+        echo "--image <image_name>       specify a container image to use"
+        echo "--count <num>              specify the number of container instances to run"
+        echo "--numa-pinning             run container with NUMA pinning"
+        exit 0
+        ;;
+    --image)
+        shift
+        if test $# -gt 0; then
+            export IMAGE_NAME=$1
+        else
+            echo "no contain image specified"
+            exit 1
+        fi
+        shift
+        ;;
+    --count)
+        shift
+        if test $# -gt 0; then
+            export COUNT=$1
+        else
+            echo "no running count specified"
+            exit 1
+        fi
+        shift
+        ;;
+    --numa-pinning)
+        export NUMA_PINNING=1
+        shift
+        ;;
+    *)
+        break
+        ;;
+    esac
+done
+
+get_cpu_pinning() {
+    sockets_num=$(lscpu | grep "Socket(s):" | sed "s/.* //g")
+
+    for cpu in $(ls -d /sys/devices/system/cpu/cpu[0-9]* | sort -t u -k 3 -n); do
+        socket_id=$(cat $cpu/topology/physical_package_id)
+        tmp_name=cpusets_socket${socket_id}
+        declare ${tmp_name}+=$(cat $cpu/topology/thread_siblings_list)" "
+    done
+
+    cpu_pinning_s=()
+    count_per_socket=$((COUNT / sockets_num))
```

Comment:
What happens when COUNT is not divisible by sockets_num...?

Reply:
line 90 will add the remainder to the last socket. For 2 sockets, 5 instances, it will assign 2 to socket 0, 3 to socket 1. For 4 sockets, 6 instances, it will be 1,1,1,3.
Thanks for pointing out this, Nitin. Will improve it to make it assign instance more evenly, make it to 1,1,2,2.

Reply:
Ok that sounds good to me Lin.